ledger/catchpointtracker.go

```diff
@@ -163,35 +163,11 @@ func (ct *catchpointTracker) initialize(cfg config.Local, paths DirsAndPrefix) {
 	// the temp file uses the hot data directories
 	ct.tmpDir = paths.HotGenesisDir
 
-	switch cfg.CatchpointTracking {
-	case -1:
-		// No catchpoints.
-	default:
-		// Give a warning, then fall through to case 0.
-		logging.Base().Warnf("catchpointTracker: the CatchpointTracking field in the config.json file contains an invalid value (%d). The default value of 0 would be used instead.", cfg.CatchpointTracking)
-		fallthrough
-	case 0:
-		if cfg.Archival && (cfg.CatchpointInterval > 0) {
-			ct.catchpointInterval = cfg.CatchpointInterval
-			ct.enableGeneratingCatchpointFiles = true
-		}
-	case 1:
-		if cfg.CatchpointInterval > 0 {
-			ct.catchpointInterval = cfg.CatchpointInterval
-			ct.enableGeneratingCatchpointFiles = cfg.Archival
-		}
-	case 2:
-		if cfg.CatchpointInterval > 0 {
-			ct.catchpointInterval = cfg.CatchpointInterval
-			ct.enableGeneratingCatchpointFiles = true
-		}
-	case forceCatchpointFileGenerationTrackingMode:
-		if cfg.CatchpointInterval > 0 {
-			ct.catchpointInterval = cfg.CatchpointInterval
-			ct.enableGeneratingCatchpointFiles = true
-			ct.forceCatchpointFileWriting = true
-		}
+	if cfg.StoresCatchpoints() {
```

Comment:
I think it still misses misses the case `cfg.CatchpointTracking=1 && cfg.Archival=False` when we do not store catchpoints but generate labels?

Reply:
Yes, it's not easily integrated into the main switch statement without multiple return values. I've decided to just move it into a separate function entirely.

Now we will have `StoresCatchpoints` and `TracksCatchpoints` which should capture all of the differences properly.